k8s/envoy-x509/README.md

```diff
@@ -262,42 +264,44 @@ Once the pod is ready, refresh the browser using the correct URL for `frontend-2
 On the other hand, you can check that the `frontend` service is still able to get a response from the `backend`. Refresh the browser at the correct URL (e.g. `http://35.222.164.221:3000`) and confirm that account details are shown for _Jacob Marley_.
 
 
-# Cleanup
+# Extend the Scenario with a Role Based Access Control Filter
 
-When you are finished running this tutorial, you can use the following script to remove all the resources used for configuring Envoy to perform X.509 authentication on workload's behalf. This command will remove:
-   * - All resources created for the SPIRE - Envoy X.509 integration tutorial.
-   * - All deployments and configurations for the SPIRE Agent, SPIRE Server, and namespace.
+Envoy provides a Role Based Access Control (RBAC) HTTP filter that checks the request based on a list of policies. A policy consists of a list of permissions and principals where the principal specifies the downstream client identities of the request, for example, the URI SAN of the downstream client certificate. So we can use the SPIFFE ID assigned to the service to create policies that allow for more granular access control.
+
+the `Symbank` demo application consumes three different endpoints to get all the information about the bank account. The `/profiles` endpoint is used to get the name and the address of the account's owner. The other two endpoints, `/balances` and `/transactions` retrieves the balance and the list of transactions of the account.
+
+Then it is possible to create a policy that allows the `frontend` service to obtain only data for profiles and denied the request sent to other endpoints. This is achieved by defining a policy which principal matches the SPIFFE ID of the service and the permissions allows only GET requests to the `/profiles` resource.
```

Comment:
I've made some adjustments here. We may need to chat about how to fix what I've broken.

```suggestion
Envoy provides a Role Based Access Control (RBAC) HTTP filter that checks the request based on a list of policies. A policy consists of permissions and principals, where the principal specifies the downstream client identities of the request, for example, the URI SAN of the downstream client certificate. So we can use the SPIFFE ID assigned to the service to create policies that allow for more granular access control.

The `Symbank` demo application consumes three different endpoints to get all the information about the bank account. The `/profiles` endpoint provides the name and the address of the account's owner. The other two endpoints, `/balances` and `/transactions`, provide the balance and transactions for the account.

To demonstrate an Envoy RBAC filter, we can create a policy that allows the `frontend` service to obtain only data from the `/profile` endpoint and deny requests sent to other endpoints. This is achieved by defining a policy with a principal that matches the SPIFFE ID of the service and the permissions to allow only GET requests to the `/profiles` resource.
```